docker-compose.yml

```diff
@@ -23,13 +23,12 @@ services:
     command: yarn dev
     build:
       context: .
-      dockerfile: Dockerfile
-      args:
-        pull: 1
+      dockerfile: Dockerfile.dev
     ports:
       - "3000:3000"
     volumes:
-      - .:/opt/outline:cached
+      - .:/opt/outline
+      - /opt/outline/node_modules
```

Comment:
As I understand it moving `node_modules` to a volume means that you're proposing running `yarn build` outside of docker in development?

If that's the case this wouldn't be acceptable as it requires the host machine to have the same version of node installed

Reply:
> requires the host machine to have the same version of node installed

It does not - but I will describe in more detail what is happening and see if that answers the question better.

The listing of node modules as a volume without a source (notice no `:`), is a bit of a trick to ensure that the node_modules _of the image_ is used. This is necessary otherwise when you mount the source code into the container for dev it would _not_ have a node modules folder and you wouldn't be able to start the app.

I did not do the same thing for the _dist_ folder since the dev Dockerfile workflow doesn't produce a dist folder. However, should you choose to manually run `yarn build` in the docker environment, the dist folder would end up on the host machine and accessible to the user's dev tools should they want to look at these files in their editor. I can add `/opt/outline/dist` to the volumes list and it would _capture_ the folder within the container in a manner similar to how `/opt/outline/node_modules` is handled. Easy to change to whichever you prefer.

In either case - the user does not run `yarn build` or any other commands on the host machine - everything happens within the docker environment. `make up` was updated to drop the user into a docker shell from where they are able to run `yarn dev` or iterate with other commands.